Assets/Plugins/Source/Editor/IConfigEditor.cs

```diff
@@ -22,18 +22,56 @@
 
 namespace PlayEveryWare.EpicOnlineServices
 {
-    //-------------------------------------------------------------------------
+    using System.IO;
+
+    
     // Interface for allowing adding additional config files to the Config editor
-    public interface IPlatformSpecificConfigEditor
+    public interface IConfigEditor
     {
-        string GetNameForMenu();
-
-        void Awake();
+        string GetName();
 
-        void LoadConfigFromDisk();
+        void Read();
 
-        void SaveToJSONConfig(bool prettyPrint);
+        void Save(bool prettyPrint);
 
         void OnGUI();
     }
+
+    public abstract class ConfigEditor<T> : IConfigEditor where T : IEmpty, ICloneableGeneric<T>, new()
```

Comment:
It's a little unconventional to have the interface and an implementing abstract class in the same file - but this practice exists across the project, and I figured that resolving that pattern should be done in one PR that doesn't also include functional changes.